packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-logic.ts

```diff
@@ -117,19 +119,19 @@ export async function createSignSendProcessDRRequests({
     bulkBodyData: drRequestsGatewayV2,
     samlCertsAndKeys,
   });
-  const responses = await sendSignedDRRequests({
-    signedRequests,
-    samlCertsAndKeys,
-    patientId,
-    cxId,
+
+  const resultPromises = signedRequests.map(async (signedRequest, index) => {
+    return sendProcessRetryDrRequests({
+      signedRequest,
+      samlCertsAndKeys,
+      patientId,
+      cxId,
+      index,
+    });
   });
-  const results = await Promise.all(
-    responses.map(async response => {
-      return await processDrResponse({
-        drResponse: response,
-      });
-    })
-  );
+
+  const results = await Promise.all(resultPromises);
```

Comment:
before we awaited all promises to resolve at each async step. Now, each send -> process -> retry is its own promise based chain of events independent of all the other chains of events